modules/drivers/peripherals/x9c/x9c.js

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (c) 2023  Moddable Tech, Inc.
+ *
+ *   This file is part of the Moddable SDK Runtime.
+ *
+ *   The Moddable SDK Runtime is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU Lesser General Public License as published by
+ *   the Free Software Foundation, either version 3 of the License, or
+ *   (at your option) any later version.
+ *
+ *   The Moddable SDK Runtime is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU Lesser General Public License for more details.
+ *
+ *   You should have received a copy of the GNU Lesser General Public License
+ *   along with the Moddable SDK Runtime.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/*
+    Renesas X9C Series - Digital Potentiometers - X9C102, X9C103, X9C104, X9C503
+    https://www.renesas.com/us/en/products/analog-products/data-converters/digital-controlled-potentiometers-dcp/x9c104-digitally-controlled-potentiometer-xdcp
+    Datasheet: https://www.renesas.com/us/en/document/dst/x9c102-x9c103-x9c104-x9c503-datasheet
+    Reference Driver: https://github.com/lucyamy/LapX9C10X/blob/main/src/LapX9C10X.cpp
+*/
+
+
+import Timer from "timer"
+
+
+class X9C
+{
+    #incrementPin;
+    #upDownPin;
+    #csNVRAMWritePin;
+    #currentWiper;
+    #minWiperValue = 0;
+    #maxWiperValue = 99;
+
+    constructor(options)
+    {   
+        const { increment, upDown, csNVRAMWrite } = options;
+        try
+        {
+            this.#incrementPin = new increment.io({
+                mode: increment.io.Output,
+                pin: increment.pin
+            });
+            this.#upDownPin = new upDown.io({
+                mode: upDown.io.Output,
+                pin: upDown.pin
+            });
+            this.#csNVRAMWritePin = new csNVRAMWrite.io({
+                mode: csNVRAMWrite.io.Output,
+                pin: csNVRAMWrite.pin
+            });
+        }
+        catch (e)
+        {
+            this.close();
+            throw e;
+        }
+
+        this.#reset(1);
```

Comment:
It would be great if one could pass in an initial value to avoid glitching to 1 if that's not the desired init value...